src/world/script_api/7E0E80.c

```diff
@@ -185,7 +194,124 @@ ApiStatus func_80280410(Evt* script, s32 isInitialCall) {
     return ApiStatus_DONE2;
 }
 
-INCLUDE_ASM(ApiStatus, "world/script_api/7E0E80", ShowShopPurchaseDialog, Evt* script, s32 isInitialCall);
+ApiStatus ShowShopPurchaseDialog(Evt* script, s32 isInitialCall) {
+    PlayerData* playerData = &gPlayerData;
+    GameStatus* gameStatus = gGameStatusPtr;
+    s32 shopItemSlot = script->varTable[0];
+    Shop* shop = gameStatus->mapShop;
+    StaticInventoryItem* shopInventory = &shop->staticInventory[shopItemSlot];
+    StaticItem* shopItem = &gItemTable[shopInventory->itemID];
+    ShopOwner* shopOwner;
+    Evt* shopOwnerScript;
+    EvtSource* shopOwnerScriptSource;
+    s32 bpCost;
+    s32 args;
+
+    shop->flags &= ~SHOP_FLAGS_1;
+    func_800E9900();
+    if (isInitialCall) {
+        D_80286530 = 0;
+        D_80286534 = 0;
+        bpCost = -1;
+        if (shopItem->typeFlags & ITEM_TYPE_FLAGS_40) {
+            bpCost = D_8008F072[shopItem->moveID * 0x14];
```

Comment:
This looks suspiciously like an array of structs size 0x14, and from the star rod database:
`58:dat : 8008F060 : MoveTable : move_table_entry[185`]`

so actually, this should be `gMoveTable[shopItem->moveID].costBP`